lib/repl.js

```diff
@@ -347,7 +351,8 @@ function REPLServer(prompt,
         if (process.domain) {
           debug('not recoverable, send to domain');
           process.domain.emit('error', err);
-          process.domain.exit();
+          if (process.domain)
+            process.domain.exit();
```

Comment:
Can you explain why these changes became necessary? It’s not obvious from looking at the diff…

Reply:
When emitting `uncaughtException` while being in an active domain, the domain will be set to null. That results in error here. I worked around it now by just emitting on next tick.